### ISSUE TITLE
Consider doing another PHP dependency bump to PHP 8.2+

### ISSUE THREAD

**Issue by Seldaek** [composer/composer#12177]
This would mean a few things:

- Drop PHP 7.2 to 8.1 support in Composer 2.9+
- Make Composer 2.8 LTS along with 2.2, meaning longer bugfix support
- Allow us to upgrade to Symfony components v7.2 (latest)
- Leaves 36% of "users" behind currently [per the stats](https://packagist.org/php-statistics) - This is a bit larger a number than I'd like currently, but I hope 8.1 will drop soon especially with the appealing stuff coming up in 8.4, and PHP 7.4 will probably stay as is for a long time but we can't support it forever.

I appreciate any thoughts/feedback here.

**Comment by alcohol** [comment]
My first and only thought was; darn, I will need to rethink the docker versioning strategy again if we're going to have multiple LTS versions. But that's trivial in comparison to the other issues.

**Comment by Seldaek** [comment]
Yeah I wouldn't worry too much there, if you go just use LTS without version number (2.2 is also available) you're kinda asking for trouble IMO. But also at this point if LTS jumps from 2.2 to 2.8 it would probably not cause many issues as the pre-7.2 PHP versions are really a tiny percentage.

**Comment by michael-o** [comment]
Please consider that the [maintenance timeframe for 8.1 has been extended recently](https://www.php.net/supported-versions.php). Since you want to bump PHP req hard to 8.x you maybe should also bump 3.0.

**Comment by stof** [comment]
@michael-o I don't see how changing the version to 3.0 is relevant here. It would only make it harder by forcing plugins to add support for a new major version while no BC breaks are planned.

The min PHP requirement impacts which version of composer you should install on your system. If you use the official composer installer, it already takes the PHP version into account, thanks to the endpoint at https://getcomposer.org/versions that provides the mapping between channels and the downloadable phars, with PHP requirements taken into account (the `min-php` field matches the format of the `PHP_VERSION_ID` constant).
And if you download manually using one of the "latest *" [download URLs](https://getcomposer.org/download/#manual-download), bumping to 2.8 or 3.0 will still impact you as 3.0 would be the latest stable and the latest preview and 3.x would be the latest snapshot

**Comment by alcohol** [comment]
> Please consider that the [maintenance timeframe for 8.1 has been extended recently](https://www.php.net/supported-versions.php). Since you want to bump PHP req hard to 8.x you maybe should also bump 3.0.

That's security patches only, hardly relevant really.

**Comment by michael-o** [comment]
> > Please consider that the [maintenance timeframe for 8.1 has been extended recently](https://www.php.net/supported-versions.php). Since you want to bump PHP req hard to 8.x you maybe should also bump 3.0.
> 
> That's security patches only, hardly relevant really.

That's correct, but do you expect that people upgrade constantly?

**Comment by michael-o** [comment]
> Yes.. Upgrading regularly is the right thing to do IMO, and what a lot of people are doing per the stats https://packagist.org/php-statistics
> 
> But if you can't for some reason then that's ok too, and that's why I'd keep maintaining the old version as LTS for bugfixes, much like I've don with 2.2.

I see, some people like me try to avoid the burden to upgrade every 6 to 12 months. We run a minor release for a couple of years.

**Comment by Seldaek** [comment]
Sure, but I'd say that puts you on 8.2 that's already two yers old now.. Or maybe 8.1 but then it's 3yo and really about time to upgrade. So by the time we get that release out I'd consider it very reasonable to require 8.2.

**Comment by michael-o** [comment]
> Sure, but I'd say that puts you on 8.2 that's already two yers old now.. Or maybe 8.1 but then it's 3yo and really about time to upgrade. So by the time we get that release out I'd consider it very reasonable to require 8.2.

What is the ETA of this change? Next year?

**Comment by Seldaek** [comment]
Not sure but yeah probably late this year or early next. In any case AFTER the 8.4 release, so that'd leave us supporting the last 3 minor PHP versions.

**Comment by alexisLefebvre** [comment]
> Not sure but yeah probably late this year or early next. In any case AFTER the 8.4 release, so that'd leave us supporting the last 3 minor PHP versions.

Time have passed since this message, maybe you can proceed to the next step now?

**Comment by Seldaek** [comment]
Yeah I've just been busy with other things and this isn't extremely urgent so it'll probably be some more time sorry. On the upside, supporting PHP 8.2+ now means 75% of users are able to upgrade, which is better.

**Comment by stof** [comment]
@Seldaek another benefit of this bump: currently `composer completion` supports only `bash` due to using symfony/console 5.4. Newer versions have support for `zsh` and `fish` as well.

**Comment by stof** [comment]
I suggest targeting PHP 8.2+, which covers 85% of the usages stats of packagist.org now, and allows using Symfony 7.4 LTS.
Bumping to 8.3 as minimum does not provide extra value (we would still be stuck at Symfony 7.4, while dropping part of the userbase) and bumping to PHP 8.4 keeps only 47% of the usages.

**Comment by Seldaek** [comment]
Yeah 8.2+ is the plan as per the title :) And yeah I'm aware of the many new features in symfony/console, but anyway many other things got in the way of this so it'll still be a bit.
